adobe/spectrum-web-components
# feat(picker): support responsive delivery of menu

## Description
Add support for delivering the options in a Picker, Action Menu, or Split Button in an `sp-tray` element when on "mobile".
- leverage `MatchMediaController` to centralize the usage/manage of `window.matchMedia` interactions
- choose between the `sp-popover` and `sp-tray` element as appropriate when opening the Menu element
  - currently DOES NOT change delivery while the Menu is open
- correct the open/close timing in Tray for use in an overlay
- surface additional style management in Menu

## Motivation and context
More complete coverage of the responsive delivery of the Spectrum spec.

## How has this been tested?

    1. Go [here](https://popover-tray--spectrum-web-components.netlify.app/)
    2. Size your browser at more that 700px wide
    3. Open a Picker element
    4. See it in a Popover
    1. Go [here](https://popover-tray--spectrum-web-components.netlify.app/)
    2. Size your browser at less that 700px wide
    3. Open a Picker element
    4. See it in a Popover
    1. Using a device that is less than 700px wide or tall and leverages a coarse pointer and does not allow hover, a phone.
    2. Go [here](https://popover-tray--spectrum-web-components.netlify.app/)
    3. Open a Picker element
    4. See it in a Popover

## Types of changes

## Checklist
